examples/bin/start-druid-main

```diff
@@ -0,0 +1,436 @@
+    memory_for_druid = int(system_memory / (1024 * 1024))
+    return memory_for_druid
+
+def convert_total_memory_string(memory):
+    try:
+        if memory == "":
+            computed_memory = compute_system_memory()
+            return computed_memory
+        elif memory.endswith(MEMORY_MEGABYTES_IDENTIFIER):
+            return int(memory[:-1])
+        elif memory.endswith(MEMORY_GIGABYTES_IDENTIFIER):
+            return 1024 * int(memory[:-1])
+        else:
+            error_and_exit('Incorrect format for memory argument, expected format is <integer_value><m/g>')
+    except Exception:
+        error_and_exit('Incorrect format for memory argument, expected format is <integer_value><m/g>')
+
+def check_memory_constraint(total_memory, service_list):
+    # 80% of total memory >= sum of lower bound service memory should be
+    lower_bound_memory = 0
+
+    for service in service_list:
+        lower_bound_memory += SERVICE_MEMORY_LOWER_BOUND.get(service)
+
+    required_memory = int(lower_bound_memory / 0.8)
+
+    if total_memory < required_memory:
+        error_and_exit(f'Minimum memory required for starting services is {required_memory}m')
+
+    if total_memory >= 2 * lower_bound_memory:
+        return int(total_memory / 2)
+    else:
+        return lower_bound_memory
+
+def build_mm_task_javaopts_array(memory_type):
+    task_memory = f'{MM_TASK_JAVAOPTS_PROP}=['
+
+    MEM_ARRAY = MM_TASK_MEM_MAP.get(memory_type)
+
+    javaopts_list = MM_TASK_JAVAOPTS_ARRAY + MEM_ARRAY
+
+    for item in javaopts_list:
+        task_memory += f'\"{item}\";'
+
+    task_memory = task_memory[:-1]
+    task_memory += ']'
+    return task_memory
+
+def build_memory_config_string(service, allocated_memory):
+    if service == MM_TASK:
+        if allocated_memory >= 2048:
+            task_count = int(allocated_memory / 2048)
+            memory_type = MM_TASK_MEMORY_TYPE_HIGH
+            task_memory = 2048
+        else:
+            task_count = int(allocated_memory / 512)
+            memory_type = MM_TASK_MEMORY_TYPE_LOW
+            task_memory = 512
+        task_count = min(task_count, multiprocessing.cpu_count())
+
+        javaopts_array = build_mm_task_javaopts_array(memory_type)
+        return [f'{MM_TASK_WORKER_CAPACITY_PROP}={task_count}', javaopts_array], task_memory * task_count
+    else:
+        heap_memory = SERVICE_MEMORY_HEAP_PERCENTAGE.get(service) * allocated_memory
+        direct_memory = int(allocated_memory - heap_memory)
+        heap_memory = int(heap_memory)
+
+        if direct_memory == 0:
+            return f'-Xms{heap_memory}m -Xmx{heap_memory}m', allocated_memory
+
+        return f'-Xms{heap_memory}m -Xmx{heap_memory}m -XX:MaxDirectMemorySize={direct_memory}m', allocated_memory
+
+def distribute_memory_over_services(services, total_memory):
+    service_memory_config = {}
+
+    memory_weight_sum = 0
+
+    service_list = services.copy()
+    if MM in services:
+        service_list.append(MM_TASK)
```

Comment:
shouldn't this take into account the number of tasks?

Reply:
Individual task memory is fixed, either 512m or 2g (heap + direct memory). 

This part of the code computes the total memory for all tasks and using that it computes the number of tasks.